pytpcc/drivers/mongodbdriver.py

```diff
@@ -1149,6 +1149,6 @@ def get_server_status(self):
     def save_result(self, result_doc):
         self.result_doc.update(result_doc)
         self.result_doc['after']=self.get_server_status()
-        self.client.test.results.save(self.result_doc)
+        self.client.test.results.insert_one(self.result_doc)
```

Comment:
I don't think this is correct, save is either an insert or a replace. The logic above only mimics the first part.

Should this be an upsert (see [here](https://github.com/mongodb/mongo-python-driver/blob/24cc4c42bf6476c25f90cfcc7e10e63e359465b2/doc/migrate-to-pymongo4.rst#collectionsave-is-removed))

Reply:
@jimoleary That is correct, it does formally change the logic. The literal replacement would be  with the whole code you linked above. However as far as I was looking into what is going on here, it appears that only insert happens here - there is nothing to update. Unless I missed something here, why we should process update_one logic here if it never happens?

Reply:
I'm not as familiar with this code so I'll defer to you. But can we add a comment to say that this is the case?

Reply:
added a comment - let me know if it helps

Reply:
Comment looks good. I don't believe we ever set an explicit _id field in result_doc, so this should always be an insert.

Reply:
I'm also not a fan of saving the results back into the database itself, but that's a separate issue.